### DISCUSSION TITLE
Insertions represented as duplications and problems with genomic coordinates.

### DISCUSSION THREAD

**Discussion by andreasprlic** [HGVSnomenclature/hgvs-nomenclature#165] [HGVS nomenclature proposals]
Biological sequences are highly repetitive and this feature I believe is at the core of the [hgvs prioritization recommendation](https://hgvs-nomenclature.org/stable/recommendations/general/) which states that insertions should be represented as duplications where possible. While I find this useful for protein sequences, I would like to demonstrate a complication around that when working with chromosomal sequences:

## A challenging insertion variant

If you look at the reference sequence for this (intentionally incorrectly represented) variant: `NC_000005.10:g.123346517_123346518insATTA` , we can see that the reference chromosome here has an `ATTA`, `ATTA` repeat (forward facing strand in both GRCh37/38). This variant is the insertion of a third such repeat unit. There is a stop codon on the reverse facing transcript in this repetitive sequence. 

<img width="1007" alt="Screenshot 2024-04-15 at 10 53 34 AM" src="https://github.com/HGVSnomenclature/hgvs-nomenclature/assets/187599/15e296de-a2a5-4456-a7be-196786b65695">

When working with samples and calling this variant with a variant caller, it would come out as an insertion of `ATTA` in a VCF file. Following the prioritization rules (which don’t mention repeats) would lead us to represent this insertion as a duplication: `NC_000005.10:g.123346522_123346525dup`. This unfortunately moves the duplicated region well into the protein coding region of the exon, rather than in the UTR, if we were to use left-shuffled genomic coordinates for the insertion. 

Biologically, the impact of this variant is not going to be coding, so the 3’ shuffling rule is not helpful here and actually causing confusion. The duplication nomenclature does not help in understanding the consequence of the insertion. The inserted sequence may be more relevant than the fact that this is a duplication.

Technically, a change from an insertion to a duplication changes the underlying nature of a variant. What is a point (insertion), becomes a range (duplication). This makes it more difficult to use in projections onto coding and protein sequences.

In transcript space we could describe this either as `NM_153223.3:c.*1_*2insTAAT` or `NM_153223.3:c.2959_*1dup`. 

## Proposed change to hgvs recommendations: 

I would like to propose two changes to the hgvs recommendations:

1)  The prioritization rules for chromosomal sequences should get changed and NO LONGER require that insertions in repetitive sequences be represented as duplications.  This would  apply only to “small” variants. Copy Number Variants, or any variants where it is impractical to represent the full insertion sequence, can still be represented as dups.

2) I’d also like to propose that a left-shuffled (5’ end) base representation of the genomic variant should become allowable as well in cases such as the above example. There’s already an exception for the 3’ rule around intron exon boundaries in coding DNA, and I’d like to propose an exception for genomic coordinates: genomic variants should get represented in the form that impacts the protein the least. I.e. in the above example we would use the representation that moves the insertion into the  UTR region (5’ in forward facing coordinates).

**Comment by jfjlaros** [comment]
Thank you for your proposal.

Here are my first thoughts.

- In my opinion, the prioritisation rules are complex enough as is, adding more exceptions will make the HGVS nomenclature harder to use. I would rather see we drop exceptions than to add them.
- I read the paragraph [you refer to](https://github.com/HGVSnomenclature/hgvs-nomenclature/blob/0712fff484d9b3b304527b558ff497115f913a45/docs/background/numbering.md?plain=1#L22-L26) as follows: "we are not allowed to ignore intronic sequences when applying the 3' rule". I think therefore that we should label this paragraph as a note rather than an exception.
- The direction in which genomic variants are shifted should be independent of transcript annotation. First of all to keep the complexity of the HGVS nomenclature under control (also see the first point), but also because genes may overlap and/or have antisense transcription. It may not be clear which transcript to avoid.
- The `c.` coordinate system can be used to view a variant from the perspective of a specific transcript, which seems to work reasonably well in this case: with the description `NC_000005.10(NM_153223.4):c.2959_*1dup` we indicate that the last part of the transcript was duplicated, from the last coordinate (`*1`), we see that the duplicated sequence is inserted one nucleotide after the stop codon.

**Comment by ifokkema** [comment]
I don't see obvious reasons for the request to remove the duplication prioritization; I believe the highlighted issue is because of the thought the variant's position on the genome and the variant's position on the transcript should point to the same position in the aligned sequences. However, this is an incorrect assumption. A logical one, when looking at the sequence alignment, but not when looking at the HGVS descriptions of the variant. As Jeroen already indicated, the random shifting of the variant on the genome level will cause issues with overlapping genes. Also, more importantly, this creates even more options to describe the same variant, which is already a big issue. This also applies to removing the duplication prioritization; allowing duplications also to be described as insertions creates more possibilities for variants to be described in multiple ways, while we need fewer instead.

When projecting variants between the genome and transcripts, they should always be re-validated by software tools that apply the 3' shifting rule correctly. As such, the given duplication, in reality, does not fully overlap the coding sequence but extends into the UTR. That already indicates that there is, most likely, no effect on the protein level.

**Comment by reece** [comment]
I agree with @jfjlaros and @ifokkema: Renormalizing a variant in the context of a transcript structure before inferring protein consequence seems likely to solve the original issue.

I recommend that we use this discussion as the basis for updating the HGVS Nomenclature to include a more complete description of projecting variants and inferring consequences, with special attention to corner cases like variants that potentially shuffle across intron-exon and exon-UTR boundaries.

**Comment by andreasprlic** [comment]
Thanks for the comments. I agree, the variant is best described in terms of the transcript. Talking some more about the _genomic_ representation of this event: I feel a bit uncomfortable to call an event that is [ATTA]x2 -> [ATTA]x3 a duplication. 

Should this event perhaps get represented as a repeat? The [prioritization rules](https://github.com/HGVSnomenclature/hgvs-nomenclature/blob/cb8bf27dedacad5a7b3daecdcf30650a63389f77/docs/recommendations/general.md?plain=1#L56) don't say anything about the relationship between repeats and duplications. Is this something that is worth some more clarification?  

I tried to find more guidance on the boundaries between repeats and duplications and noticed that [in this example](https://github.com/HGVSnomenclature/hgvs-nomenclature/blob/cb8bf27dedacad5a7b3daecdcf30650a63389f77/docs/recommendations/DNA/repeated.md?plain=1#L36) they are presented as possible alternatives.

**Reply by ifokkema** [reply]
You're right; prioritization rules on duplications/expanded repeats and deletions/contracted repeats are missing. We have some other open issues related to repeat regions as well:
https://hgvs-nomenclature.org/stable/consultation/open-issues/#repeated-sequences

While it's important that identical variants are described the same way, the subject of repeat sequences strongly divides groups; some really like it, and some really hate it. While we're internally discussing how this can be solved and better standardized, feel free to describe this variant, _on the genomic level_, as a repeat expansion: `NC_000005.10:g.123346518_123346525ATTA[3]`. For coding transcripts, repeat syntax can only be used for trinucleotide repeats, so on the transcript level, you'll have to use the duplication syntax.
